rust/private/rustc.bzl

```diff
@@ -1046,6 +1046,9 @@ def rustc_compile_action(
         files = getattr(ctx.files, "data", []) + coverage_runfiles,
         collect_data = True,
```

Comment:
https://docs.bazel.build/versions/main/skylark/lib/ctx.html#runfiles and https://docs.bazel.build/versions/main/skylark/rules.html#runfiles-features-to-avoid discourage use of `collect_data`

Reply:
Do you want to attempt to remove it in this change? Or an isolated change?

Reply:
I'm not really sure I understand the implications of using "collect_data" here, so I left it alone.